### PR TITLE
Add cleanup support to library

### PR DESCRIPTION
**Pull request by juancoen**
- Also refactor and comment API Sync library interface

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -7,34 +7,88 @@ var APISync = function () {
...
+   */
   function getAllWorkspaces() {
     return container.get('workspaceRepository').getAll();
   }
 
-  function setup(directory, accessToken, strategy) {
-    setWorkingDirectory(directory);
-    setContext(accessToken, directory);
+  /**
+   * Cleanups the specified directory configuration.
```
Comment (dmartinezg) [inline lib/index.js]: `Cleans up`

Comment (dmartinezg) [inline lib/index.js]: What is a 'clean-up' BTW?

File: lib/index.js [hunk 2]
```diff
@@ -7,34 +7,88 @@ var APISync = function () {
...
+      return container.get('cleanupController').cleanup();
+    });
+  }
 
-    return container.get('setupController').setup(strategy);
+  /**
+   * Runs the setup for the specified directory using the specified strategy.
+   *
+   * @param  {String} directory The path of the user directory.
+   * @param  {String} accessToken A valied CS access token.
```
Comment (dmartinezg) [inline lib/index.js]: Typo `valid`

File: lib/index.js [hunk 3]
```diff
@@ -7,34 +7,88 @@ var APISync = function () {
...
 
-    return container.get('pullController').getAPIFiles(accessToken);
+  /**
+   * Returns the status for the specified directory.
+   *
+   * @param  {String} directory The path of the user directory.
+   * @param  {String} accessToken A valid CS access token.
+   * @return {Promise} A Promise that will return an object containing the
+   *                  status of all the files in the
+   *                  directory and the exitings conflicts with the Anypoint
```
Comment (dmartinezg) [inline lib/index.js]: `existing` (?)

File: lib/index.js (hunk 1 above)
Comment (juancoen) [inline lib/index.js]: Fixed and added more details to the doc. The cleanup removes the configuration for the specified directory. This includes the selected Business Group and API Version.

File: lib/index.js (hunk 2 above)
Comment (juancoen) [inline lib/index.js]: fixed in all places

File: lib/index.js (hunk 3 above)
Comment (juancoen) [inline lib/index.js]: fixed

**Comment by dmartinezg** [comment]
:+1:

**Pull request merged**